lib/quandl/model/database.rb

```diff
@@ -0,0 +1,61 @@
+module Quandl
+  class Database < ModelBase
+    include Quandl::Operations::Get
+    include Quandl::Operations::List
+
+    def datasets(options = {})
+      Quandl::Dataset.all({ params: { database_code: database_code, query: nil, per_page: Quandl::ApiConfig.default_per_page, page: 1 } }.deep_merge(options))
+    end
+
+    def bulk_download_url(options = {})
+      options.assert_valid_keys(:download_type)
+
+      url = Quandl::ApiConfig.api_base + '/' + self.class.default_path + '/data'
+      url = Quandl::Util.constructed_path(url, id: database_code)
+
+      params = {}
+      params['download_type'] = options[:download_type] if options[:download_type]
+      params['auth_token'] = Quandl::ApiConfig.api_key if Quandl::ApiConfig.api_key
+
+      url += '?' + params.to_query if params.any?
+      url
+    end
+
+    def bulk_download_to_file(file_or_folder_path, options = {})
+      fail(QuandlError, 'You must specific a file handle or folder to write to.') if file_or_folder_path.blank?
+
+      # Retrieve the location of the bulk download url
+      url = bulk_download_url(options)
+      download_url = RestClient.get(url) do |response, _request, _result, &_block|
```

Comment:
consider moving this to the connection layer, in which case we need `bulk_download_to_file` to have an option not to prefix the base url.